pennylane/ops/qubit/matrix_ops.py

```diff
@@ -332,3 +334,134 @@ def _controlled(self, control):
+    >>> A = [[0.1,0.2],[0.3,0.4]]
+    >>> @qml.qnode(dev)
+    ... def example_circuit():
+    ...     qml.BlockEncode(A, wires=range(2))
+    ...     return qml.state()
+    ...     print(qml.matrix(example_circuit)())
+    [[ 0.1         0.2         0.97283788 -0.05988708]
+    [ 0.3         0.4        -0.05988708  0.86395228]
+    [ 0.94561648 -0.07621992 -0.1        -0.3       ]
+    [-0.07621992  0.89117368 -0.2        -0.4       ]]
+    """
+
+    num_params = 1
+    num_wires = AnyWires
+
+    def __init__(self, a, wires, do_queue=True, id=None):
+        a = pnp.atleast_2d(a)
+        wires = Wires(wires)
+        if pnp.sum(a.shape) <= 2:
+            normalization = a if a > 1 else 1
+            subspace = (1, 1, 2 ** len(wires))
+        else:
+            normalization = pnp.max(
+                [norm(a @ pnp.conj(a).T, ord=pnp.inf), norm(pnp.conj(a).T @ a, ord=pnp.inf)]
+            )
+            subspace = (*a.shape, 2 ** len(wires))
+
+        a = a / normalization if normalization > 1 else a
+
+        if subspace[2] < (subspace[0] + subspace[1]):
+            raise ValueError(
+                f"Block encoding a {subspace[0]} x {subspace[1]} matrix "
+                f"requires a hilbert space of size at least "
+                f"{subspace[0] + subspace[1]} x {subspace[0] + subspace[1]}."
+                f" Cannot be embedded in a {len(wires)} qubit system."
+            )
+
+        super().__init__(a, wires=wires, do_queue=do_queue, id=id)
+        self.hyperparameters["norm"] = normalization
+        self.hyperparameters["subspace"] = subspace
+
+    @staticmethod
+    def compute_matrix(*params, **hyperparams):
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.BlockEncode.matrix`
+
+        Args:
+            params (list): trainable parameters of the operator, as stored in the ``parameters`` attribute
+            hyperparams (dict): non-trainable hyperparameters of the operator, as stored in the ``hyperparameters`` attribute
+
+
+        Returns:
+            tensor_like: canonical matrix
+
+        **Example**
+
+        >>> a = np.array([[0.1,0.2],[0.3,0.4]])
+        >>> a
+        tensor([[0.1, 0.2],
+                [0.3, 0.4]])
+        >>> qml.BlockEncode.compute_matrix(a,subspace=[2,2,4])
+        array([[ 0.1       ,  0.2       ,  0.97283788, -0.05988708],
+               [ 0.3       ,  0.4       , -0.05988708,  0.86395228],
+               [ 0.94561648, -0.07621992, -0.1       , -0.3       ],
+               [-0.07621992,  0.89117368, -0.2       , -0.4       ]])
+        """
+        a = params[0]
+        n, m, k = hyperparams["subspace"]
+
+        if qml.math.sum(qml.math.shape(a)) <= 2:
```

Comment:
In this case `A` won't be a matrix?

Reply:
Yup, in this case `A` is a scalar. We can block encode scalars as the top left element of the matrix.